cwltool/builder.py

```diff
@@ -109,67 +109,43 @@ def get_requirement(self,
 
 class Builder(HasReqsHints):
     def __init__(self,
-                 job,                       # type: Dict[Text, Union[Dict[Text, Any], List, Text, None]]
-                 files=None,                # type: List[Dict[Text, Text]]
-                 bindings=None,             # type: List[Dict[Text, Any]]
-                 schemaDefs=None,           # type: Dict[Text, Dict[Text, Any]]
-                 names=None,                # type: Names
-                 requirements=None,         # type: List[Dict[Text, Any]]
-                 hints=None,                # type: List[Dict[Text, Any]]
-                 timeout=None,              # type: float
-                 debug=False,               # type: bool
-                 resources=None,            # type: Dict[str, int]
-                 js_console=False,          # type: bool
-                 mutation_manager=None,     # type: Optional[MutationManager]
-                 formatgraph=None,          # type: Optional[Graph]
-                 make_fs_access=None,       # type: Type[StdFsAccess]
-                 fs_access=None,            # type: StdFsAccess
-                 force_docker_pull=False,   # type: bool
-                 loadListing=u"",           # type: Text
-                 outdir=u"",                # type: Text
-                 tmpdir=u"",                # type: Text
-                 stagedir=u"",              # type: Text
-                 job_script_provider=None   # type: Optional[Any]
+                 job,                         # type: Dict[Text, Union[Dict[Text, Any], List, Text, None]]
+                 timeout,                     # type: float
+                 files=[],                    # type: List[Dict[Text, Text]]
+                 bindings=[],                 # type: List[Dict[Text, Any]]
+                 schemaDefs={},               # type: Dict[Text, Dict[Text, Any]]
+                 names=Names(),               # type: Names
+                 requirements=[],             # type: List[Dict[Text, Any]]
+                 hints=[],                    # type: List[Dict[Text, Any]]
```

Comment:
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
Good catch, wasn't aware of the behaviour: I've removed all lists and dicts as default arguments